portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,40 @@ const randomFactGenerator = () => {
     const factContainer = document.getElementById('fact-container');
     factContainer.innerText = randomFact;
 };
+
+/**
+ * Fetches a the content from the server and adds it to the DOM.
+ */
+const getServletContent = () => {
+    console.log('Fetching the content from the server.');
+
+    // The fetch() function returns a Promise because the request is asynchronous.
+    const responsePromise = fetch('/data');
+
+    // When the request is complete, pass the response into handleResponse().
+    responsePromise.then(handleResponse);
+};
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+const handleResponse = response => {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addContentToDOM);
+};
+
+/** Adds a random quote to the DOM. */
+const addContentToDOM = content => {
+  console.log('Adding content to dom: ' + content);
+
+  const quoteContainer = document.getElementById('servlet-content');
+  quoteContainer.innerText = content;
```

Comment:
Don't gotta address this for the tutorial, but for future reference: inserting raw content this way could be dangerous, [exposing your app to XSS attacks!](https://stackoverflow.com/questions/52707031/does-innertext-prevent-xss)

Reply:
But this would be ameliorated by using React or Angular lol

Reply:
How exactly would this occur. To be honest, I sort of just followed the example  code they used.

Reply:
No, you did fine, dude-- tbh, I'm not sure how big the risk is, since you used `innertext` instead of `innerhtml`. It's just that whenever I see direct DOM injection, my spidey-sense flares up, and I thought I'd warn you for your capstone or future projects!